File: P1/ej4/filters/filter_chain.py
class FilterChain:
    def __init__(self):
        self.filters = []
        self.target = None
    
    def add_filter(self, filter_obj):
        self.filters.append(filter_obj)
        return self
    
    def set_target(self, target):
        self.target = target

    def execute(self, request):
        errors = []
        for filter_obj in self.filters:
            error = filter_obj.execute(request)
            if error:
                errors.append(error)
        return errors

File: P2/ejercicio1/lib/filters/filter_chain.dart
import 'base.dart';

class FilterChain {
  List<Filter> filters = [];
  dynamic target;

  FilterChain addFilter(Filter filterObj) {
    filters.add(filterObj);
    return this;
  }

  void setTarget(dynamic target) {
    this.target = target;
  }

  List<String> execute(Map<String, String> request) {
    List<String> errors = [];
    for (var filterObj in filters) {
      String? error = filterObj.execute(request);
      if (error != null) {
        errors.add(error);
      }
    }
    return errors;
  }
}